master/internal/db/cache.go

```diff
@@ -0,0 +1,187 @@
+package db
+
+import (
+	"context"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/determined-ai/determined/master/pkg/archive"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+)
+
+type modelDefFolder struct {
+	fileTree   []*experimentv1.FileNode
+	cachedTime time.Time
+	lock       sync.RWMutex
+	path       string
+}
+
+// FileCache is metadata for files cached at file system.
+type FileCache struct {
+	rootDir string
+	maxAge  time.Duration
+	caches  map[int]*modelDefFolder
+	lock    sync.Mutex
+}
+
+// NewFileCache initialize FileCache obj.
+func NewFileCache(rootDir string, maxAge time.Duration) *FileCache {
+	return &FileCache{
+		rootDir: rootDir,
+		maxAge:  maxAge,
+		caches:  make(map[int]*modelDefFolder),
+	}
+}
+
+func (f *FileCache) getOrCreateFolder(expID int) (*modelDefFolder, error) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	value, ok := f.caches[expID]
+	if !ok {
+		exp := struct {
+			ModelDefinition []byte
+		}{}
+		err := Bun().NewSelect().TableExpr(
+			"experiments").Column("model_definition").Where("id = ?", expID).Scan(context.TODO(), &exp)
```

Comment:
what is the backend-eng team's stance on context.TODO?

It seems to me like it is easy to pass the real request's context into this code.  I would imagine that we should do that here.

Reply:
Thank you for catching this, will be fixed